### PR TITLE
Add content view syntax highlighting for JSON and msgpack

### LINKED ISSUE #5623: mitmweb: Add support for syntax highlighting / coloring
#### Problem Description
When selecting "View: msgpack" on a msgpack request or response, the resultant JSON is displayed using single-quotes (`'`), which is not valid. It should use double-quotes (`"`) instead.

#### Steps to reproduce the behavior:
1. Have a request/response in `mitmweb` and click Replace and use the file contained in this archive: [get_resource_version.zip](https://github.com/mitmproxy/mitmproxy/files/9692381/get_resource_version.zip)
2. Select "View: msgpack" next to the Replace button
3. Notice that the resultant JSON contains single-quoted properties and strings:
```json
{
    'data_headers': {
        'result_code': 1
    },
    'data': {
        'resource_version': 'y2XM6giU6zz56wCm'
    }
}
```

I am not an expert on the JSON standard, but pasting the output into a [JSON validator](https://duckduckgo.com/?q=json+validator&t=ffab&ia=answer) throws up errors which can be fixed by using double quotes (and I assume that's what the red highlighting means, since the above code block starts with ```json)

#### System Information
```
>mitmproxy --version
Mitmproxy: 8.1.1
Python:    3.10.6
OpenSSL:   OpenSSL 1.1.1n  15 Mar 2022
Platform:  Windows-10-10.0.22622-SP0
```

#### P.S.
I know a bit of Python but am unfamiliar with this codebase, however I would be happy to try fixing this myself and submitting a PR if that would be easier? It seems like something that would be simple enough for a first time contributor to attempt.

### PR DESCRIPTION
**Pull request by SapiensAnatis**
#### Description

Adds support for syntax highlighting in the mitmweb content view for JSON and msgpack bodies. Closes #5623

Visually speaking, they look identical in the content view, except msgpack supports a few types which JSON does not. These are left unhighlighted, but the formatting function should be easy to extend to cover these if desired.

image

Summary of changes:

 - Modified existing JSON token labelling function to distinguish keys from regular strings
 - Modified msgpack formatting function to add token labelling 
 - Added stylesheet colours for msgpack and json colours 
     - I just picked a few random colours that looked half-decent, but I haven't given too much thought to accessibility/colourblindness. Open to discussion on changing these.
 - Modified test_json.py and test_msgpack.py to test the output of the format function

#### Checklist

 - [x] I have updated tests where applicable.
 - [x] I have added an entry to the CHANGELOG.

### CODE REVIEW

**Comment by SapiensAnatis** [comment]
re: pipeline failures -- unsure of what caused `individual_coverage` to fail, but it appears that  `mypy` is failing because of the msgpack formatting function not having an unconditional return.

I found this function quite difficult to put together, because it was recursive in nature but had to produce a list which sets the content in the view line-by-line. It was difficult to avoid nesting within the lines, particularly when the data contained nested lists/objects. To accomplish this, when the function 'bottoms out' on a basic property (int/string/bool/whatever), it just modifies a parameter 'output' array rather than returning. I'm sure someone else could propose a better way of writing this function, but that's what I ended upm with.

It might be possible to fix it by adding dummy return statements, whose values are never used, in each branch. But this feels like bad practice to me. Let me know if you have any suggestions on how best to address this.

**Comment by SapiensAnatis** [comment]
Thanks for the review! I think my latest commits should address both of your comments. 

One thing I have thought of which I'm not sure about, though, is whether msgpack dicts could have keys that are not strings. I have been making sample msgpack responses using ASP.NET and [MessagePack-CSharp](https://github.com/neuecc/MessagePack-CSharp), which only allows you to set strings or ints as keys to an object (and when using ints, the response is interpreted as a list rather than a dict by mitmproxy). However I can't find anything in the MessagePack spec that restricts what type the keys in a map can be.

Currently, keys are rendered as `"key"` which works for the responses I was able to make with the aforementioned setup.

File: mitmproxy/contentviews/msgpack.py [hunk 1]
```diff
@@ -15,23 +15,62 @@ def parse_msgpack(s: bytes) -> Any:
...
+        token = [("Token_Keyword_Constant", repr(data))]
+        output[-1] += token
+
+        return output
+
+    elif type(data) is dict:
+        output[-1] += [("text", "{")]
+        for key in data:
+            output.append([indent, ("text", "    "), ("Token_Name_Tag", f'"{key}"'), ("text", ": ")])
+            _ = format_msgpack(data[key], output, indent_count + 1)
```
Comment (mhils) [inline mitmproxy/contentviews/msgpack.py:47]: Skip unnecessary `_ = ` here?

File: mitmproxy/contentviews/msgpack.py [hunk 2]
```diff
@@ -44,7 +83,7 @@ class ViewMsgPack(base.View):
     def __call__(self, data, **metadata):
         data = parse_msgpack(data)
         if data is not PARSE_ERROR:
-            return "MsgPack", format_msgpack(data)
+            return "MsgPack", (line for line in format_msgpack(data))
```
Comment (mhils) [inline mitmproxy/contentviews/msgpack.py:86]: Contentviews are expected to return an iterator over `TViewLine`. Just returning `format_msgpack(data)` as-is should be fine as lists are iterable. :)

**Review by mhils — APPROVED** [review]
Looks excellent now. 👍 

> One thing I have thought of which I'm not sure about, though, is whether msgpack dicts could have keys that are not strings.

To clarify, that already affects the existing implementation? In this case I'd be happy to go ahead and merge this. If you want to investigate msgpack more closely that can be done in a separate PR.

**Comment by SapiensAnatis** [comment]
> To clarify, that already affects the existing implementation? In this case I'd be happy to go ahead and merge this. If you want to investigate msgpack more closely that can be done in a separate PR.

No, it just might look a bit weird -- if it is indeed possible to have arbitrary types as keys it might render like this for a boolean key:

```json
{
    "true": 1
}
```

i.e. the current code assumes keys are strings and so surrounds them with double quotes (as `repr(key)` produces the single quotes that got me started on these msgpack formatting changes :P ). Not even sure what the desired behaviour in that case would be, so I think you're right that it's probably best left as a separate discussion.

Will make the code changes you requested shortly.

**Comment by mhils** [comment]
Awesome, thanks! 🍰

**Pull request merged**